python/cudf_polars/cudf_polars/containers/column.py

```diff
@@ -117,3 +120,44 @@ def nan_count(self) -> int:
                 plc.DataType(plc.TypeId.INT32),
             )
         ).as_py()
+
+
+class NamedColumn(Column):
+    """A column with a name."""
+
+    name: str
+
+    def __init__(
+        self,
+        column: plc.Column,
+        name: str,
```

Comment:
Just confirming polars column labels just supports `str` and not (generically) `abc.Hashable`?

Reply:
yep

Reply:
Any validation needed before assigning `self.name = name` or is that guaranteed elsewhere (besides mypy)?

Reply:
Guaranteed by construction pretty much.

A bit of a difference is that this isn't "user-facing" code, so mypy guarantees are much stronger.